app/src/main/java/Craigslist/clone/App.java

```diff
@@ -21,6 +23,41 @@ public static void main(String[] args) {
             result.put("total_listing_count", connector.getNumListings());
             ctx.json(result);
         });
+
+        app.get("/listings/all", ctx -> {
+            DBConnector connector = DBConnector.get();
+            List<Listing> temp = connector.getAllListings();
+            ctx.json(temp);
+        });
+
+        app.get("/listings/create", ctx -> {
+           DBConnector connector = DBConnector.get();
+           Listing listing = ctx.bodyAsClass(Listing.class);
+           connector.createListing(listing);
+        });
+
+        app.get("/user/create", ctx -> {
+            DBConnector connector = DBConnector.get();
+            User user = ctx.bodyAsClass(User.class);
+            connector.createUser(user);
+        });
+
+        app.get("/listings/delete/{listingID}", ctx -> {
+            DBConnector connector = DBConnector.get();
+            connector.deleteListing(Integer.parseInt(ctx.pathParam("listingID")));
```

Comment:
here and every where you are using `Integer.parseInt`, can you catch the exception if the parsing fails, something like this:
```
  try {
    int val = Integer.parseInt(text);
  } catch (NumberFormatException e) {
     // we need to return status code `400` to the user since it is a bad request.
    ctx.status(400);
  }
```

Learn more about http status codes here: [400 Bad Request](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400) and here: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status.